File: Kotlin/primes.kt
fun main(args: Array<String>) {
	print(2u)
	val limit = args[0].toUInt()
	val half = (limit / 2u).toInt()
	var (sieve, total) = BooleanArray(half) { true } to 1u
	for (i in 3..kotlin.math.sqrt(limit.toDouble()).toInt() step 2)
		if (sieve[i / 2])
			for (j in i * i / 2 until half step i)
				sieve[j] = false
	for (i in 1 until half)
		if (sieve[i]) {
			print(", ${2u * i.toUInt() + 1u}")
			total++
		}
	println("\nTotal: $total")
}

File: Swift/primes.swift
print(terminator: "2")
let limit = UInt32(CommandLine.arguments[1])!, half = Int(limit / 2)
var sieve = [Bool](repeating: true, count: half), total = 1
for i in stride(from: 3, through: Int(Float(limit).squareRoot()), by: 2) {
	if sieve[i / 2] {
		for j in stride(from: i * i / 2, to: half, by: i) {
			sieve[j] = false
		}
	}
}
for i in stride(from: 1, to: half, by: 1) {
	if sieve[i] {
		print(",", 2 * i + 1, terminator: "")
		total += 1
	}
}
print("\nTotal:", total)